pkg/deploy/api/v1beta3/types.go

```diff
@@ -67,9 +67,15 @@ type CustomDeploymentStrategyParams struct {
 // RecreateDeploymentStrategyParams are the input to the Recreate deployment
 // strategy.
 type RecreateDeploymentStrategyParams struct {
+	// TimeoutSeconds is the time to wait for updates before giving up. If the
+	// value is nil, a default will be used.
+	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty" description:"the time to wait for updates before giving up"`
```

Comment:
Why is this specific to strategies? Open an issue that we should move it in spec. Also I believe nil should mean I don't want to gate my deployments with timeouts.